src/sentry/integrations/discord/client.py

```diff
@@ -77,9 +115,17 @@ def send_message(
         """
         Send a message to the specified channel.
         """
-        self.post(
-            MESSAGE_URL.format(channel_id=channel_id),
-            data=message.build(notification_uuid=notification_uuid),
-            timeout=5,
-            headers=self.prepare_auth_header(),
-        )
+        if not self.is_bot_blocked_by_overrides(channel_id=channel_id):
```

Comment:
do we want to check this for every message? for this, this can really be slow and unnecessary for most of cases.

Can we do this instead?
1. Always try to post message
2. Check error code and verify it's permission error
3. Mark this an user error and bump a "user error counter". With this, we have flexibility to only monitor "non user-error" calls.